lib/pos_blockchain.go

```diff
@@ -74,12 +146,17 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
 	if err = bc.storeValidatedBlockInBlockIndex(desoBlock); err != nil {
 		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem adding block to block index: ")
 	}
+	newBlockNode, exists := bc.blockIndex[*blockHash]
+	if !exists {
+		return false, false, nil, errors.New("processBlockPoS: Block not found in block index after adding it. " +
+			"This should never happen.")
+	}
 
 	// 4. Try to apply the incoming block as the new tip. This function will
 	// first perform any required reorgs and then determine if the incoming block
 	// extends the chain tip. If it does, it will apply the block to the best chain
 	// and appliedNewTip will be true and we can continue to running the commit rule.
-	appliedNewTip, err := bc.tryApplyNewTip(desoBlock, currentView, lineageFromCommittedTip)
+	appliedNewTip, err := bc.tryApplyNewTip(newBlockNode, currentView, lineageFromCommittedTip)
```

Comment:
Is this correct? Assuming that the current block's parent is validated and connected from the above processBlock(block.Parent) call, then this tryApplyNewTip just adds the current block to the tip of the chain. It's not possible to reorg here because the recursive calls to processBlock(...) perform their reorgs internally.

Maybe I'm missing something?

Reply:
It's possible that the parent blocks will not satisfy the `shouldReorg` condition 
`blockNode.Header.ProposedInView >= currentView`, in which case we will not reorg.

Reply:
I'm concerned about the recursive process parent calls above tho. Does it break fast hotstuff rules in any way? Does the reorg condition need to be updated to require that the ProposedInView of this block must exceed the current uncommitted tip's ProposedInView?

Reply:
This is resolved by the recursive validateAndIndexBlockPoS function in the downstream PRs.